### PR TITLE
Feature/espresso seq

### PR DESCRIPTION
**Pull request by fabiooshiro**
Espresso seq implementation (alpha)

```shell
git checkout feature/espresso-seq
go build
./nonodo --sequencer=espresso --from-block=98299 --namespace=10008 --enable-debug
```
PS.: it still needs an implementation of the message sender and other metadata

### CODE REVIEW

File: internal/rollup/espresso_api.go
Comment (ZzzzHui) [inline internal/rollup/espresso_api.go:1]: There are 6 typos in this file. `ExpressoService` -> `EspressoService`

**Comment by ZzzzHui** [comment]
I'm wondering whether this line is using a different namespace value https://github.com/Calindra/nonodo/blob/9deb29a5d169c0a9c643cd8cd30ece505131563b/internal/rollup/espresso_api.go#L102
while in other places, the value is passed in from `NonodoOpts`
https://github.com/Calindra/nonodo/blob/9deb29a5d169c0a9c643cd8cd30ece505131563b/internal/nonodo/nonodo.go#L229

**Comment by ZzzzHui** [comment]
Currently, type `EspressoAPI` is not used 
https://github.com/Calindra/nonodo/blob/9deb29a5d169c0a9c643cd8cd30ece505131563b/internal/rollup/espresso_api.go#L15
If it's used in the future, the 3rd function declaration in this type may need to be updated.

**Comment by ZzzzHui** [comment]
Listening on testnet works great. But submitting to testnet doesn't seem to work. Got the tx hash but no tx appears on their explorer. I'm investigating 
https://github.com/Calindra/nonodo/blob/9deb29a5d169c0a9c643cd8cd30ece505131563b/internal/sequencers/espresso/espresso_listener_test.go#L34

File: internal/rollup/espresso_api.go
Comment (sandhilt) [reply internal/rollup/espresso_api.go:1]: PR https://github.com/Calindra/nonodo/pull/54

**Comment by sandhilt** [comment]
> Currently, type `EspressoAPI` is not used
> 
> https://github.com/Calindra/nonodo/blob/9deb29a5d169c0a9c643cd8cd30ece505131563b/internal/rollup/espresso_api.go#L15
> 
> If it's used in the future, the 3rd function declaration in this type may need to be updated.

PR https://github.com/Calindra/nonodo/pull/56

**Comment by sandhilt** [comment]
> I'm wondering whether this line is using a different namespace value
> 
> https://github.com/Calindra/nonodo/blob/9deb29a5d169c0a9c643cd8cd30ece505131563b/internal/rollup/espresso_api.go#L102
> 
> while in other places, the value is passed in from `NonodoOpts`
> 
> https://github.com/Calindra/nonodo/blob/9deb29a5d169c0a9c643cd8cd30ece505131563b/internal/nonodo/nonodo.go#L229

In older versions, `namespace` was `VM_ID` .
For while, we use in two scenarios:
1. We use in `gio` endpoint using `VM_ID` is always first 8 bytes of application address (maybe change this behavior later).
https://github.com/Calindra/nonodo/blob/221d9ede83492dd36554048076a93de5d27779c5/internal/rollup/rollup.go#L41
https://github.com/Calindra/nonodo/blob/221d9ede83492dd36554048076a93de5d27779c5/internal/dataavailability/espresso.go#L21
2. In Sequencer we can pass via CLI.

**Review by sandhilt — APPROVED** [review]
LGTM 👍🏽

**Comment by fabiooshiro** [comment]
Install brunodo
```shell
npm i -g brunodo@2.1.0
```

Run with Espresso as a Sequencer
```shell
brunodo --sequencer=espresso --namespace=10008 --from-block=325789
```

Send a transaction
```shell
curl --location 'https://query.cappuccino.testnet.espresso.network/submit/submit' \
--header 'Content-Type: application/json' \
--data '{
    "namespace": 10008,
    "payload": "Q2FydGVzaSBSb2NrcyE="
}'
```

**Pull request merged**